File: src/main/java/net/swamphut/swampium/service/spec/file/text/TextFileWriterService.java
package net.swamphut.swampium.service.spec.file.text;

import io.reactivex.Completable;

import java.io.File;

public interface TextFileWriterService {
    /**
     * Append the text into the end of file
     * <p>
     * Create the file if not exist
     */
    Completable append(File file, String string);

    /**
     * Write the text into file
     * <p>
     * Overwrite if the file is not empty
     * Create the file if not exist
     */
    Completable write(File file, String string);
}

File: src/main/kotlin/dev/reactant/reactant/service/spec/file/text/TextFileWriterService.kt
package dev.reactant.reactant.service.spec.file.text

import io.reactivex.Completable

import java.io.File

interface TextFileWriterService {
    /**
     * Append the text into the end of file
     *
     *
     * Create the file if not exist
     */
    fun append(file: File, string: String): Completable

    /**
     * Write the text into file
     *
     *
     * Overwrite if the file is not empty
     * Create the file if not exist
     */
    fun write(file: File, string: String): Completable
}
